apps/web/src/providers/team-context.tsx

```diff
@@ -20,21 +26,46 @@ interface TeamContextType {
   isLoading: boolean;
   currentRole: "ADMIN" | "MEMBER";
   currentIsAdmin: boolean;
+  selectTeam: (teamId: number) => void
 }
 
 const TeamContext = createContext<TeamContextType | undefined>(undefined);
 
 export function TeamProvider({ children }: { children: React.ReactNode }) {
   const { data: teams, status } = api.team.getTeams.useQuery();
+  const client = useQueryClient()
 
-  const currentTeam = teams?.[0] ?? null;
+  const [currentTeam, setCurrentTeam] = useState<Team | null>(null)
+
+  function selectTeam(teamId: number){
+    const teamFounded = teams?.find(team => team.id === teamId);
+
+    Cookies.set('unsendTeamId', String(teamId), { expires: 7, path: "/", sameSite: "lax"  })
```

Comment:
tbh, this should not be a browser side thing, should send a request to be and it should verify team and roles etc and set it as http secure cookie